test/AWSMetadataUtilities.jl

```diff
@@ -122,6 +122,36 @@ end
     @test result == expected_result
 end
 
+@testset "_uri_cleaning" begin
```

Comment:
```suggestion
@testset "__clean_uri" begin
```
I think all of these need to changed?

Reply:
Can you add a test for replacing `+` with `""`?